prompting-client/src/daemon/server.rs

```diff
@@ -42,38 +45,78 @@ macro_rules! map_enum {
     };
 }
 
-pub fn new_server_and_listener<T: ReplyToPrompt + Clone>(
-    client: T,
+pub fn new_server_and_listener<R, S>(
+    client: R,
+    reload_handle: S,
     active_prompt: ReadOnlyActivePrompt,
     tx_actioned_prompts: UnboundedSender<ActionedPrompt>,
     socket_path: String,
-) -> (AppArmorPromptingServer<Service<T>>, UnixListener) {
-    let service = Service::new(client.clone(), active_prompt, tx_actioned_prompts);
+) -> (AppArmorPromptingServer<Service<R, S>>, UnixListener)
+where
+    R: ReplyToPrompt + Clone,
+    S: SetLogLevel,
+{
+    let service = Service::new(
+        client.clone(),
+        reload_handle,
+        active_prompt,
+        tx_actioned_prompts,
+    );
     let listener = UnixListener::bind(&socket_path).expect("to be able to bind to our socket");
 
     (AppArmorPromptingServer::new(service), listener)
 }
 
-pub struct Service<R>
+pub trait SetLogLevel: Send + Sync + 'static {
+    fn set_filter(&self, level: &str) -> crate::Result<()>;
+}
+
+impl<L, S> SetLogLevel for Arc<Handle<L, S>>
+where
+    L: From<EnvFilter> + Send + Sync + 'static,
+    S: 'static,
+{
+    fn set_filter(&self, filter: &str) -> crate::Result<()> {
+        info!(?filter, "attempting to update logging filter");
+        let f = filter
+            .parse::<EnvFilter>()
+            .map_err(|_| Error::UnableToUpdateLogLevel {
+                reason: format!("{filter:?} is not a valid logging filter"),
+            })?;
```

Comment:
It looks like `EnvFilter` is defined in another crate (not in this repo) so I definitely think there should be some documentation (both in the code and in the CLI args) about what valid log levels are.